shared/utils/il2cpp-utils.hpp

```diff
@@ -83,6 +83,13 @@ namespace il2cpp_utils {
     template<CreationType creationType = CreationType::Temporary>
     Il2CppString* newcsstr(std::u16string_view inp) {
         il2cpp_functions::Init();
+
+        // if null string input,
+        // return an empty allocated il2cpp string
+        if (inp.data() == nullptr) {
+            return newcsstr<creationType>("");
```

Comment:
Use `u""` here to ensure we don't call through to the string_view one and then back over to the utf16 one